### ISSUE TITLE
Parse status stop at 99%

### ISSUE THREAD

**Issue by lpeter218065** [yasp-dota/yasp#1057]
Seems I get the error when I upload the same demo file again.
I do not understand why the job 2 start before job 1 finish.
I could not have a good way to debug on this, do you have any
suggestions ?

Thanks in advance.

Log from requests:
requests-6 waiting for parse job 1, currently in active
requests-6 waiting for parse job 1, currently in active
requests-6 connecting redis://127.0.0.1:6379/0
requests-6 connecting postgresql://postgres:postgres@localhost/yasp
requests-6 waiting for parse job 2, currently in waiting
requests-6 waiting for parse job 2, currently in waiting
requests-6 waiting for parse job 2, currently in waiting
requests-6 waiting for parse job 2, currently in waiting

Logs from parser:

parser-12 time spent on post-processing match : 4869.532ms
parser-12 saving uploaded parse
parser-12 generating player analysis: 486.355ms
parser-12 connecting postgresql://postgres:postgres@localhost/yasp
parser-12 connecting redis://127.0.0.1:6379/0
parser-12 parse job: 2
parser-12 {"url":"http://localhost:5200/redis/ec4d5cea78ed377c43cabc2ecb340b86","state":{"time":{"elapsed":2.422,"remaining":263.443},"speed":189325.3509496284,"percentage":0.009109897171598106,"size":{"total":50334926,"transferred":458546}}}
parser-12 {"url":"http://localhost:5200/redis/ec4d5cea78ed377c43cabc2ecb340b86","state":{"time":{"elapsed":3.42,"remaining":288.541},"speed":172402.9239766082,"percentage":0.01171389424512117,"size":{"total":50334926,"transferred":589618}}}
parser-12 {"url":"http://localhost:5200/redis/ec4d5cea78ed377c43cabc2ecb340b86","state":{"time":{"elapsed":4.42,"remaining":221.946},"speed":222360.63348416288,"percentage":0.019525885465690365,"size":{"total":50334926,"transferred":982834}}}

**Comment by lpeter218065** [comment]
p.s.

the logs all come from 2nd time uploading the demo file, after 1st time parsing, I restart everything including the redis server.

**Comment by lpeter218065** [comment]
And parser job continues increasing

**Comment by howardchung** [comment]
might be caused by the second upload overwriting the first.  Since we take the hash of the uploaded file as the redis key, weird things might happen if the same file is uploaded simultaneously.

**Comment by lpeter218065** [comment]
I did not upload the demo file simultaneously.

First upload is ok. then after the analysis page come out, I go to the request page and upload the file again, this time it did not give the correct result.

**Comment by howardchung** [comment]
I'm still not sure what the problem is.

You're saying the first upload is fine, but the second upload is stuck at 99%?

**Comment by lpeter218065** [comment]
yes

**Comment by howardchung** [comment]
and that's the end of the log?  I don't see any errors

**Comment by lpeter218065** [comment]
I am trying to figure out the problem.
But normally we can see from log: parse job 1, in completed (sth like that) right? this case, seems parser job 2 created and start.

**Comment by howardchung** [comment]
You said it got stuck at 99%, but the progress trace only shows 1% downloaded?

**Comment by lpeter218065** [comment]
Parse job 1 corresponds to 2nd upload. After I finish 1st upload, I restart everything including redis-server, that is why you see the 2nd parse have job number of 1.  Here job 2 is created for which I am weird about. 

Today I tried on another environment  and could not reproduce it. I guess I did some other things. I would temporarily close it.   If I meet again, I will reopen it.

Thanks for your help:D
